core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -603,6 +683,15 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  private def fetchOffsetForTimestamp(logManager: LogManager, topicPartition: TopicPartition, timestamp: Long) : Option[TimestampOffset] = {
+    logManager.getLog(TopicAndPartition(topicPartition.topic, topicPartition.partition)) match {
+      case Some(log) =>
+        log.fetchOffsetsByTimestamp(timestamp)
+      case _ =>
```

Comment:
It should be `None` here so that the exhaustiveness checking can work (i.e. if you then added an `if` to the `Some` case, it would warn you that the pattern match is not exhaustive).